plugins/terminal.koplugin/main.lua

```diff
@@ -293,13 +293,11 @@ function Terminal:editName(item)
 end
 
 function Terminal:deleteShortcut(item)
-    local shortcuts = {}
-    for _, element in ipairs(self.shortcuts) do
-        if element.text ~= item.text and element.commands ~= item.commands then
-            table.insert(shortcuts, element)
+    for i, element in ipairs(self.shortcuts) do
+        if element.text == item.text and element.commands == item.commands then
+            table.remove(self.shortcuts, i)
```

Comment:
Sure this is ok, ipairs & table.remove ?
Usually, we're either break'ing after the first remove, or iterating in reverse (`for i=#table, 1, -1 do`).

Reply:
Oh, right, of course ;o).